File: hexameter-core/src/test/java/org/codetome/hexameter/core/api/HexagonOrientationTest.java
package org.codetome.hexameter.core.api;

import org.codetome.hexameter.core.testutils.TestUtils;
import org.junit.Test;

import static junit.framework.Assert.assertEquals;
import static org.codetome.hexameter.core.api.HexagonOrientation.FLAT_TOP;
import static org.codetome.hexameter.core.api.HexagonOrientation.POINTY_TOP;

public class HexagonOrientationTest {

    @Test
    public void testEnum() {
        TestUtils.superficialEnumCodeCoverage(HexagonOrientation.class);
    }

    @Test
    public void shouldProperlyCalculateFlatCoordinateOffset() {
        assertEquals(0.0f, FLAT_TOP.getCoordinateOffset());
    }

    @Test
    public void shouldProperlyCalculatePointyCoordinateOffset() {
        assertEquals(0.5f, POINTY_TOP.getCoordinateOffset());
    }
}

File: mixite2.core/src/commonTest/kotlin/org/hexworks/mixite2/core/api/HexagonOrientationTest.kt
package org.hexworks.mixite2.core.api

import org.hexworks.mixite2.core.api.HexagonOrientation.*
import kotlin.test.Test
import kotlin.test.assertEquals

class HexagonOrientationTest {

    @Test
    fun shouldProperlyCalculateFlatCoordinateOffset() {
        assertEquals(expected = 0.0f, actual = FLAT_TOP.coordinateOffset)
    }

    @Test
    fun shouldProperlyCalculatePointyCoordinateOffset() {
        assertEquals(expected = 0.5f, actual = POINTY_TOP.coordinateOffset)
    }
}
